src/store/userOrganizationReducer.ts

```diff
@@ -0,0 +1,55 @@
+import { createSlice } from '@reduxjs/toolkit'
+import organizationService from '../api/organizationService'
+import { changeOrganization } from './organizationReducer'
+import { setSuccessNotification, setErrorNotification } from './notificationReducer'
+import Organization from '../types/organization'
+import { SetUserOrganizationState, UpdateUserOrganizationState } from './types/userOrganizationState'
+import { Id } from '../types/propertyTypes'
+import { AppDispatch } from './store'
+
+type UserOrganization = {
+    data: Organization | null
+}
+
+const initialState: UserOrganization = {
+    data: null,
+}
+
+// toolkit sets up the redux and state
+const organizationSlice = createSlice({
+    name: 'userOrganization',
+    initialState,
+    reducers: {
+        setUserOrganization(_state: UserOrganization, action: SetUserOrganizationState) {
+            return {
+                data: action.payload
+            }
+        },
+        updateUserOrganization(state: UserOrganization, action: UpdateUserOrganizationState) {
+            return {
+                ...state,
+                data: {
+                    ...(state?.data || {}),
+                    ...action.payload
+                }
```

Comment:
i am pretty sure you dont need the `?` in `state?.data` on line 32 since `state` will be defined. initial state is `{ data: null }`

its fine to leave it if it seems like it could be undefined though